apps/src/code-studio/components/ShareDialog.jsx

```diff
@@ -160,94 +190,129 @@ var ShareDialog = React.createClass({
                     <i className="fa fa-twitter"></i>
                   </a>}
-               </span>}
+                </span>}
+              </div>
+              {this.state.showSendToPhone &&
+              <SendToPhone
+                channelId={this.props.channelId}
+                appType={this.props.appType}
+                styles={{label:{marginTop: 15, marginBottom: 0}}}
+              />}
+              {(this.props.appType === 'applab' || this.props.appType === 'gamelab') &&
+              <AdvancedShareOptions
+                i18n={this.props.i18n}
+                shareUrl={this.props.shareUrl}
+                onClickExport={this.props.onClickExport}
+                expanded={this.state.showAdvancedOptions}
+                onExpand={this.showAdvancedOptions}
+                channelId={this.props.channelId}
+                embedOptions={embedOptions}
+              />}
+              {/* Awkward that this is called continue-button, when text is
+               close, but id is (unfortunately) used for styling */}
+              <button
+                id="continue-button"
+                style={{position: 'absolute', right: 0, bottom: 0, margin: 0}}
+                onClick={this.close}
+              >
+                {this.props.i18n.t('project.close')}
+              </button>
             </div>
-            {this.state.showSendToPhone &&
-             <SendToPhone
-               channelId={this.props.channelId}
-               appType={this.props.appType}
-               styles={{label:{marginTop: 15, marginBottom: 0}}}
-             />}
-            {(this.props.appType === 'applab' || this.props.appType === 'gamelab') &&
-             <AdvancedShareOptions
-               i18n={this.props.i18n}
-               shareUrl={this.props.shareUrl}
-               onClickExport={this.props.onClickExport}
-               expanded={this.state.showAdvancedOptions}
-               onExpand={this.showAdvancedOptions}
-               channelId={this.props.channelId}
-               embedOptions={embedOptions}
-             />}
-            {/* Awkward that this is called continue-button, when text is
-            close, but id is (unfortunately) used for styling */}
-            <button
-              id="continue-button"
-              style={{position: 'absolute', right: 0, bottom: 0, margin: 0}}
-              onClick={this.close}
-            >
-              {this.props.i18n.t('project.close')}
-            </button>
           </div>
-        </div>
-      </BaseDialog>
+        </BaseDialog>
+        <PublishDialog/>
+      </div>
+
     );
   }
 });
-module.exports = ShareDialog;
+
+export const UnconnectedShareDialog = ShareDialog;
+
+export default connect(state => ({
+  isOpen: state.shareDialog.isOpen,
+}), (dispatch, ownProps) => ({
+  onClose: () => dispatch(hideShareDialog()),
+  onShowPublishDialog() {
+    dispatch(hideShareDialog());
+    dispatch(showPublishDialog(ownProps.channelId, ownProps.appType));
```

Comment:
cant we still avoid ownProps here and have publish call `this.props.onShowPublishDialog(this.props.channelId, this.props.appType)`?